### PR TITLE
C tests for internal machinery

### PR DESCRIPTION
**Pull request by parro-it**
I add a bunch of tests developed in C to harden the internals.
There are many more to write, but I already discovered some problem...

I tried to use [check framework](https://libcheck.github.io/check/) but without success... I guess it need an executable to run, while we start from a node addon...

We should find a way to remove all this code from the production build....

### CODE REVIEW

**Comment by mischnic** [comment]
> We should find a way to remove all this code from the production build....

Only possibility I can think of right now: Include these sources
https://github.com/parro-it/libui-napi/blob/ee835aeadc9e8e128f82335b2d091d8a0e03405e/binding.gyp#L7-L12
only if a specific environment variable is set. Add a check in the javascript test file and print an error if the functions weren't compiled in.

**Comment by parro-it** [comment]
> We should find a way to remove all this code from the production build....
> Is all that code contained in the release build? (Not that large but...)

😆

**Comment by parro-it** [comment]
I used an env var LIBUI_TARGET=test to include test sources in `tools/list-sources.js`

File: appveyor.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ test_script:
 after_test:
   - if defined APPVEYOR_REPO_TAG_NAME (
       npm run rebuild &&
-      7z a -ttar %APPVEYOR_REPO_TAG_NAME%-win-%PLATFORM%.tar node_libui.node &&
+      7z a -ttar %APPVEYOR_REPO_TAG_NAME%-win-%PLATFORM%.tar libui.node &&
```
Comment (parro-it) [inline appveyor.yml]: Unfortunately, on windows the rename does not work: macOS & linux strip 'lib' from the file name if it should normally begin with 'lib'. Eg. libui.node -> ui.node. Windows does not.

**Pull request merged**